src/components/application/slplayer.vue

```diff
@@ -0,0 +1,403 @@
+        </v-flex>
+      </v-layout>
+    </div>
+  </div>
+</template>
+
+<script>
+import shaka from 'shaka-player/dist/shaka-player.ui.debug';
+import { mapActions, mapGetters, mapMutations, mapState } from 'vuex';
+import slplayer from '@/store/modules/slplayer';
+
+import messages from '@/components/messages.vue';
+import plexthumb from './plexbrowser/plexthumb.vue';
+
+import 'shaka-player/dist/controls.css';
+shaka.log.setLevel(shaka.log.Level.WARNING);
+shaka.polyfill.installAll();
+
+import BitrateSelectionFactory from '@/player/ui/bitrateselection';
+import SubtitleSelectionFactory from '@/player/ui/subtitleselection';
+import AudioSelectionFactory from '@/player/ui/audioselection';
+import MediaSelectionFactory from '@/player/ui/mediaselection';
+import CloseButtonFactory from '@/player/ui/closebutton';
+import Forward30ButtonFactory from '@/player/ui/forward30button';
+import Replay10ButtonFactory from '@/player/ui/replay10button';
+
+export default {
+  name: 'slplayer',
+  components: {
+    plexthumb, messages,
+  },
+
+  data() {
+    return {
+      eventbus: window.EventBus,
+      dialog: false,
+      lastSentTimeline: {},
+      metadataLoadedPromise: null,
+
+      playerConfig: {
+        streaming: {
+          bufferingGoal: 120,
+        },
+      },
+
+      playerUiOptions: {
+        controlPanelElements: [
+          'replay10',
+          'play_pause',
+          'forward30',
+          'mute',
+          'volume',
+          'close',
+          'time_and_duration',
+          'spacer',
+
+          'overflow_menu',
+          'fullscreen',
+        ],
+
+        overflowMenuButtons: [
+          'picture_in_picture',
+          'cast',
+          'bitrate',
+          'subtitle',
+          'audio',
+          'media',
+        ],
+      },
+    };
+  },
+
+  beforeCreate() {
+    if (!this.$store.state.slplayer) {
+      this.$store.registerModule('slplayer', slplayer);
+    }
+  },
+
+  created() {
+    this.metadataLoadedPromise = this.FETCH_METADATA();
+    shaka.ui.OverflowMenu.registerElement('bitrate', new BitrateSelectionFactory(this.eventbus));
+    shaka.ui.OverflowMenu.registerElement('subtitle', new SubtitleSelectionFactory(this.eventbus));
+    shaka.ui.OverflowMenu.registerElement('audio', new AudioSelectionFactory(this.eventbus));
+    shaka.ui.OverflowMenu.registerElement('media', new MediaSelectionFactory(this.eventbus));
+    shaka.ui.Controls.registerElement('close', new CloseButtonFactory(this.eventbus));
+    shaka.ui.Controls.registerElement('forward30', new Forward30ButtonFactory());
+    shaka.ui.Controls.registerElement('replay10', new Replay10ButtonFactory());
+  },
+
+  async mounted() {
+    await this.metadataLoadedPromise;
+    this.SET_PLAYER(new shaka.Player(this.$refs.videoPlayer));
+    this.SET_PLAYER_CONFIGURATION(this.playerConfig);
+    this.SET_PLAYER_UI(new shaka.ui.Overlay(this.GET_PLAYER, this.$refs.videoPlayerContainer,
+      this.$refs.videoPlayer,
+    ));
+
```

Comment:
Add this here to make casting work 😄 
```
    // Get and configure cast app id.
    if (this.$refs.videoPlayerContainer['dataset'] && this.$refs.videoPlayerContainer['dataset']['shakaPlayerCastReceiverId']) {
      this.playerUiOptions.castReceiverAppId = this.$refs.videoPlayerContainer['dataset']['shakaPlayerCastReceiverId'];
    }
```

Reply:
Got a slightly more elegant implementation adding this in the UI options directly (just need to ensure that cast js is loaded by then)